java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java

```diff
@@ -491,6 +494,45 @@ public void visitLambdaExpression(LambdaExpressionTree tree) {
         }
       }
       super.visitLambdaExpression(tree);
+      if(lambdaType.isUnknown() || lambdaType.isTagged(JavaType.DEFERRED)) {
+        return;
+      }
+      // Try to complete type of lamdba if required
+      JavaSymbol.MethodJavaSymbol samMethod = null;
+      for (Symbol member : lambdaType.symbol().memberSymbols()) {
+        if (member.isMethodSymbol() && member.isAbstract()) {
+          samMethod = (JavaSymbol.MethodJavaSymbol) member;
+        }
+      }
+      Preconditions.checkState(samMethod != null);
+      // compute lambda return type
+      JavaType samReturnType = (JavaType) samMethod.returnType().type();
+      JavaType capturedReturnType = resolve.resolveTypeSubstitution(samReturnType, lambdaType);
+      // TODO compute return type for lambda with block and return statements
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1135)